keras_nlp/src/models/gemma/gemma_attention.py

```diff
@@ -186,3 +193,35 @@ def call(
         if cache is not None:
             return attention_output, cache
         return attention_output
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "head_dim": self.head_dim,
+                "num_query_heads": self.num_query_heads,
+                "num_key_value_heads": self.num_key_value_heads,
+                "kernel_initializer": self._kernel_initializer,
+                "dropout": self.dropout,
+            }
+        )
+        # Ensure the serialziation of the dtype polices
+        dtype_policies = {
```

Comment:
Ideally, we land this for one model like gemma, but for all KerasNLP models. We want our model specific code to be fairly concise when possible, and this would add a lot of extra code to add to every model we have.

Let's think if there's a way we can do this where most of the logic is consolidated to base classes like `backbone.py`.